test/functional/feature_proxy.py

```diff
@@ -320,19 +320,34 @@ def networks_dict(d):
 
         self.stop_node(1)
 
-        self.log.info("Test passing invalid -proxy raises expected init error")
-        self.nodes[1].extra_args = ["-proxy=abc:def"]
-        msg = "Error: Invalid -proxy address or hostname: 'abc:def'"
+        self.log.info("Test passing invalid -proxy hostname raises expected init error")
+        self.nodes[1].extra_args = ["-proxy=abc:23456"]
```

Comment:
This now performs and depends on the outcome of a DNS lookup, I think. Need to ensure it always fails.

Reply:
Good catch. As per the RFC 2181: 

> The DNS itself places only one restriction on the particular labels
   that can be used to identify resource records [...] The length of
   any one label is limited to between 1 and 63 octets.

I therefore propose to set the hostname with an invalid 0-length label (e.g. `abc..abc`), which should always fail.